packages/referral_reconciliation/lib/pages/create_referral/record_reason_checklist.dart

```diff
@@ -539,98 +573,114 @@ class _ReferralReasonChecklistPageState
     } else if (item.dataType == 'String') {
       return Padding(
         padding: const EdgeInsets.all(8.0),
-        child: DigitTextField(
-          onChange: (value) {
-            checklistFormKey.currentState?.validate();
-          },
-          isRequired: item.required ?? true,
-          controller: controller[index],
-          inputFormatter: [
-            FilteringTextInputFormatter.allow(RegExp(
-              "[a-zA-Z0-9 ]",
-            )),
-          ],
+        child: FormField<String>(
           validator: (value) {
-            if (((value == null || value == '') && item.required == true)) {
+            // Custom validation logic
+            if (((controller[index].text == '') && item.required == true)) {
```

Comment:
Previously, we were using custom flutter component so it was populating the value with the textfield but now value is coming null so instead of value we are directly taking the value from form.